### DISCUSSION TITLE
cx_Oracle

### DISCUSSION THREAD

**Discussion by claudioaguiarg** [flet-dev/flet#3381] [Packages] [answered]
I can't build with this package on Windows

**Comment by ndonkoHenri** [comment, marked as answer]
`cx_Oracle` can't be supported in a Flet **mobile** app, but the good news is there's a fully working replacement that already is.

**Why `cx_Oracle` won't work on mobile:** cx_Oracle only runs in "thick" mode — it bundles ODPI-C, which `dlopen`s the proprietary **Oracle Instant Client** (`libclntsh`) at runtime to talk to the database. Oracle does **not** ship Instant Client for iOS or Android, so there is no client library for the package to load on a phone. Even if the wheel built, it could never open a connection on-device. It's also deprecated upstream — Oracle renamed/replaced it with **[python-oracledb](https://python-oracledb.readthedocs.io/)**. Use it instead (Flet now [supports](https://pypi.flet.dev/oracledb) it on both Android and iOS) and its default **thin mode** is pure-Python (Oracle Net straight over sockets) — no Oracle Client libraries required.

Add it to your dependencies:

```toml
# pyproject.toml
dependencies = ["oracledb"]
```

```python
import oracledb

conn = oracledb.connect(user="scott", password="tiger", dsn="host:1521/service_name")
cur = conn.cursor()
cur.execute("SELECT 1 FROM dual")
print(cur.fetchall())
```

The API is largely a drop-in for `cx_Oracle`. Just make sure to use Thin mode — don't call `oracledb.init_oracle_client()` (that switches to Thick mode, which has the same Instant Client problem).

Let know if you face issues.
